docs: document new settings

maxz opened the pull request:
When bringing my `transmission-daemon` settings up to version 4.0, I had to search through the commit history to find out what some of the new settings do. I documented all those settings based on what I found and tested.
I'm not entirely certain if I missed something for `enable-tcp`, e. g. I'm not sure whether it influences the connection to the tracker.

I also fixed some minor typos.
I could not find ` tr_inactvelimit`, `tr_inactivelimit` or spelling variations thereof in the source code, so I'm also not sure whether it should even still be there.

GaryElshaw commented:
The rest of the world outside of the US does customise, or make customisations, you know? :-)

maxz commented:
That's how it was written before I got there, you know? Feel free to change it to "customise" and "behaviour".

maxz commented:
I rebased the branch onto the latest changes in main and squashed the corrections into the base commits. I had to force push thrice, because one correction did not properly manifest in the final commits. While the commits got properly squashed, that particular line got overwritten by the later kept commit again, so I just edited the commit.

ckerr commented:
Notes: Documented new settings.
